src/mem/chunkmap.h

```diff
@@ -47,12 +59,14 @@ namespace snmalloc
 // Use flat map is under a single node.
 #  define SNMALLOC_MAX_FLATPAGEMAP_SIZE PAGEMAP_NODE_SIZE
```

Comment:
Would it be better to change 
```
#  define SNMALLOC_MAX_FLATPAGEMAP_SIZE PAGEMAP_NODE_SIZE
```
to 
```
#  define SNMALLOC_MAX_FLATPAGEMAP_SIZE \
  pal_supports<LazyCommit> ? .... : PAGEMAP_NODE_SIZE
```
Where we choose `...` suitably.

Reply:
With default chunk size (i.e., neither `USE_SMALL_CHUNKS` nor `USE_LARGE_CHUNKS`), `SUPERSLAB_BITS` is 20.  With a 48-bit VA, that means a flat chunkmap pagemap is 256 MiB.  Is that a reasonable choice for `...`?

Reply:
Probably, I would say going above 2^32 causes a lot of issues, so below that maybe 1GiB is the max?

Reply:
Permitting 512M will break `src/test/func/memory/memory.cc`; we could adjust the test, tho'.

Reply:
Either is fine.  Just document the choice.